app/javascript/components/Hello.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+type MyProps = {
+  name?: string;
+}
+
+function Hello(props:MyProps) {
+  if(props.name) {
+    return <h1>Hello {props.name}</h1>
+  }
+  else {
+    return <h1>Greetings Stranger</h1>
```

Comment:
Rather than user an if else here you could look into how to set default prop values in typescript:
https://learning.oreilly.com/library/view/learn-react-with/9781789610253/7082efa2-c13d-4a0c-bfd6-5c2fc4a1b61e.xhtml

Reply:
While these files were not intended to be in this branch, I appreciate this resource!
This file will be removed now.